### PR TITLE
Pykwalify

### PR DESCRIPTION
**Pull request by erasche**
- Adds a schema
- Validates against that schema
- Removes extraneous use of travis + jenkins for just jenkins (since we have secrets involved)
- all toolshed urls are forced to `https`
- swaps a bunch of section IDs to section Labels

### CODE REVIEW

File: tools.yaml [hunk 1]
```diff
@@ -110,768 +110,768 @@ tools:
   - name: ega_download_streamer
     owner: yhoogstrate
     tool_panel_section_label: Get Data
-    tool_shed_url: toolshed.g2.bx.psu.edu
+    tool_shed_url: https://toolshed.g2.bx.psu.edu
```
Comment (bgruening) [inline tools.yaml]: We should remove the toolshed from the sections entirely. We do only support TS and not TTS ... so this should go ontop.

Comment (erasche) [reply tools.yaml]: Shall I remove:

```
- name: covenntree
  owner: steffen
  tool_panel_section_label: Graph/Display Data
  tool_shed_url: https://testtoolshed.g2.bx.psu.edu  
- name: protease_prediction
  owner: bgruening
  tool_panel_section_label: Proteomics
  tool_shed_url: https://testtoolshed.g2.bx.psu.edu 
- name: jamm
  owner: messersc
  tool_panel_section_label: Peak Calling
  tool_shed_url: https://testtoolshed.g2.bx.psu.edu 
```

Comment (bgruening) [reply tools.yaml]: yes please.
Protease prediction should be on the TS I think.

Comment (erasche) [reply tools.yaml]: cool, thanks.

**Pull request merged**